src/spec-node/singleContainer.ts

```diff
@@ -160,14 +161,24 @@ async function buildAndExtendImage(buildParams: DockerResolverParameters, config
 
 	const args: string[] = [];
 	if (buildParams.buildKitVersion) {
-		args.push('buildx', 'build',
-			'--load', // (short for --output=docker, i.e. load into normal 'docker images' collection)
-			'--build-arg', 'BUILDKIT_INLINE_CACHE=1', // ensure cache manifest is included in the image
-		);
+		if (buildParams.buildxPlatform && buildParams.buildxPush && argImageName) {
+			args.push('buildx', 'build');
+			args.push('--platform', buildParams.buildxPlatform);
+			args.push('--push');
+			args.push('-f', finalDockerfilePath, '-t', argImageName);
```

Comment:
I think we probably want the `--build-arg BUILDKIT_INLINE_CACHE=1` as per the section below so that the cache metadata is written included in the image to enable rebuilds to re-use cached layers even when not present on the build machine